pytorch_lightning/loops/base.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+from weakref import proxy
+
+import pytorch_lightning as pl
+
+
+class Loop(ABC):
+
+    def __init__(self):
+        self.iteration_count: int = 0
```

Comment:
do we need more granular counters? for example, before on_advance_start, before advance, before on_advance_end? 

let's use the tracker from the start? https://github.com/PyTorchLightning/pytorch-lightning/blob/ce976769ef2241230f13f0964a5f38dc49fdc923/pytorch_lightning/trainer/progress.py#L18-L55

Reply:
That's probably part of the internal state we wanted to introduce separately in independent refactors. This one is mainly about changing API but keeping behaviour.

Reply:
agree with Justus and I think we should definitely integrate the progress tracker for the loops. This will have to be done separately with all existing tests in mind.

Reply:
Yes, the progress tracker will be integrated promptly and replace this entirely.